core/commands/checkout.py

```diff
@@ -51,21 +45,25 @@ class GsCheckoutNewBranchCommand(WindowCommand, GitCommand):
     Prompt the user for a new branch name, create it, and check it out.
     """
 
-    def run(self):
-        sublime.set_timeout_async(self.run_async)
+    def run(self, base_branch=None):
+        sublime.set_timeout_async(lambda: self.run_async(base_branch))
 
-    def run_async(self, name=""):
-        self.window.show_input_panel(NEW_BRANCH_PROMPT, name, self.on_done, None, None)
+    def run_async(self, base_branch=None):
+        self.base_branch = base_branch
+        self.window.show_input_panel(NEW_BRANCH_PROMPT, "", self.on_done, None, None)
```

Comment:
The second argument should not an empty string. When I created this it worked like if you enter an invalid branch name, for example with a space in the middle, it would open the prompt again with the name you enter prefilled.